PyInstaller/depend/dylib.py

```diff
@@ -51,109 +60,110 @@
-    r'/libnss_files.*\.so\..*': 1,
-    r'/libnss_hesiod.*\.so\..*': 1,
-    r'/libnss_nis.*\.so\..*': 1,
-    r'/libnss_nisplus.*\.so\..*': 1,
-    r'/libresolv\.so\..*': 1,
-    r'/libutil\.so\..*': 1,
+    r'^ld-linux\.so(\..*)?$',
+    r'^libBrokenLocale\.so(\..*)?$',
+    r'^libanl\.so(\..*)?$',
+    r'^libcidn\.so(\..*)?$',
+    r'^libcrypt\.so(\..*)?$',
+    r'^libnsl\.so(\..*)?$',
+    r'^libnss_compat.*\.so(\..*)?$',
+    r'^libnss_dns.*\.so(\..*)?$',
+    r'^libnss_files.*\.so(\..*)?$',
+    r'^libnss_hesiod.*\.so(\..*)?$',
+    r'^libnss_nis.*\.so(\..*)?$',
+    r'^libnss_nisplus.*\.so(\..*)?$',
+    r'^libresolv\.so(\..*)?$',
+    r'^libutil\.so(\..*)?$',
     # libGL can reference some hw specific libraries (like nvidia libs).
-    r'/libGL\..*': 1,
-    # libxcb-dri changes ABI frequently (e.g.: between Ubuntu LTS releases) and is usually installed
-    # as dependency of the graphics stack anyway. No need to bundle it.
-    r'/libxcb\.so\..*': 1,
-    r'/libxcb-dri.*\.so\..*': 1,
-}
-
-_aix_excludes = {
-    r'/libbz2\.a': 1,
-    r'/libc\.a': 1,
-    r'/libC\.a': 1,
-    r'/libcrypt\.a': 1,
-    r'/libdl\.a': 1,
-    r'/libintl\.a': 1,
-    r'/libpthreads\.a': 1,
-    r'/librt\\.a': 1,
-    r'/librtl\.a': 1,
-    r'/libz\.a': 1,
-}
+    r'^libGL\..*',
+    # libxcb-dri changes ABI frequently (e.g.: between Ubuntu LTS releases) and
+    # is usually installed as dependency of the graphics stack anyway. No need
+    # to bundle it.
+    r'^libxcb\.so(\..*)?$',
+    r'^libxcb-dri.*\.so(\..*)?$',
+])
+
+_aix_excludes = set([
+    r'^libbz2\.a$',
+    r'^libc\.a$',
+    r'^libC\.a$',
+    r'^libcrypt\.a$',
+    r'^libdl\.a$',
+    r'^libintl\.a$',
+    r'^libpthreads\.a$',
+    r'^librt\\.a$',
+    r'^librtl\.a$',
+    r'^libz\.a$',
+])
 
 
 if is_win:
-    _includes = _win_includes
-    _excludes = _win_excludes
-    sep = '[%s]' % re.escape(os.sep + os.altsep)
-    # Exclude everything from the Windows directory by default.
-    windir = re.escape(winutils.get_windows_dir())
-    _excludes['^%s%s' % (windir, sep)] = 1
-    # Allow pythonNN.dll, pythoncomNN.dll, pywintypesNN.dll
-    _includes[r'%spy(?:thon(?:com(?:loader)?)?|wintypes)\d+\.dll$' % sep] = 1
-
+    _includes |= _win_includes
+    _excludes |= _win_excludes
 elif is_aix:
     # The exclude list for AIX differs from other *nix platforms.
-    _excludes = _aix_excludes
+    _excludes |= _aix_excludes
 elif is_unix:
     # Common excludes for *nix platforms -- except AIX.
-    _excludes = _unix_excludes
+    _excludes |= _unix_excludes
 
 
 class ExcludeList(object):
     def __init__(self):
-        self.regex = re.compile('|'.join(_excludes.keys()), re.I)
+        self.regex = re.compile('|'.join(_excludes), re.I)
 
     def search(self, libname):
         # Running re.search() on '' regex never returns None.
         if _excludes:
-            return self.regex.search(libname)
+            return self.regex.search(os.path.basename(libname))
```

Comment:
Minor enhancement: When using `match()` instead of `search()`, we could use simpler patterns: leaving of the leading `^` and in most cases the trailing `$`. I have to admit that I don't know if this speeds up notably, but at least the patterns would be easier to read :-)